internal/storage/storage.go

```diff
@@ -246,6 +271,150 @@ func (s *Storage) GetHostses() ([]Hosts, error) {
 	return hostses, err
 }
 
+// disks
+func (s *Storage) SetDisks(data string) error {
+	diskses, err := s.GetDisks()
+	if err != nil {
+		return err
+	} else if len(diskses) == 0 {
+		return s.execSQL(INSERT_DISKS, data)
+	}
+	return s.execSQL(SET_DISKS, data, diskses[0].Id)
+}
+
+func (s *Storage) GetDisks() ([]Disks, error) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	rows, err := s.db.Query(SELECT_DISKS)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	var diskses []Disks
+	var disks Disks
+	for rows.Next() {
+		err = rows.Scan(&disks.Id, &disks.Data, &disks.LastmodifiedTime)
+		diskses = append(diskses, disks)
+		break
+	}
+	return diskses, err
+}
+
+// disk
+func (s *Storage) SetDisk(host, device, mount, containerImag string, format_percent int) error {
+	disks, err := s.GetDisk(SELECT_DISK_BY_DEVICE_PATH, host, device)
+	if err != nil {
+		return err
+	} else if len(disks) == 0 {
+		return s.execSQL(
+			INSERT_DISK,
+			host,
+			device,
+			comm.DISK_DEFAULT_NULL_SIZE,
+			comm.DISK_DEFAULT_NULL_URI,
+			mount,
+			format_percent,
+			containerImag,
+			comm.DISK_DEFAULT_NULL_CHUNKSERVER_ID,
+		)
+	}
+	return s.execSQL(SET_DISK, mount, format_percent, containerImag, disks[0].Id)
+}
+
+func (s *Storage) UpdateDiskURI(host, device, devUri string) error {
+	return s.execSQL(SET_DISK_URI, devUri, host, device)
+}
+
+func (s *Storage) UpdateDiskSize(host, device, size string) error {
+	return s.execSQL(SET_DISK_SIZE, size, host, device)
+}
+
+func (s *Storage) UpdateDiskChunkServerID(host, mountPoint, chunkserverId string) error {
+	_, err := s.GetDiskByDiskFormatMountPoint(host, mountPoint)
+	if err != nil {
+		return err
+	}
+	return s.execSQL(SET_DISK_CHUNKSERVER_ID, chunkserverId, host, mountPoint)
+}
+
+func (s *Storage) DeleteDisk(host, device string) error {
+	if len(device) > 0 {
+		return s.execSQL(DELETE_DISK_HOST_DEVICE, host, device)
+	} else {
+		return s.execSQL(DELETE_DISK_HOST, host)
+	}
+}
+
+func (s *Storage) GetDiskByDiskFormatMountPoint(host, mountPoint string) (Disk, error) {
+	var disk Disk
+	disks, err := s.GetDisk(SELECT_DISK_BY_DISK_FORMAT_MOUNTPOINT, host, mountPoint)
+	if len(disks) == 0 {
+		return disk, errno.ERR_DATABASE_EMPTY_QUERY_RESULT.
+			F("The disk[host=%s, disk_format_mount_point=%s] was not found in database.",
+				host, mountPoint)
+	}
+	disk = disks[0]
+	return disk, err
+}
+
+func (s *Storage) CleanDiskChunkServerId(serviceId string) error {
+	disks, err := s.GetDisk("service", serviceId)
+	if err != nil {
+		return err
+	}
+
+	for _, disk := range disks {
+		if err := s.UpdateDiskChunkServerID(
+			disk.Host,
+			disk.MountPoint,
+			comm.DISK_DEFAULT_NULL_CHUNKSERVER_ID,
+		); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *Storage) GetDisk(filter string, args ...interface{}) ([]Disk, error) {
+	var query string
+	switch filter {
+	case "all":
```

Comment:
You can provide some const variables to refer those strings (`all`、`host`、`service`).